src/main/controlMod.F90

```diff
@@ -269,11 +269,14 @@ subroutine control_init(dtime)
     ! flag for SSRE diagnostic
     namelist /clm_inparm/ use_SSRE
 
+    ! flag for SSRE diagnostic
+    namelist /clm_inparm/ use_SSRE
```

Comment:
Should be fixed now. Do you have any expert suggestions on how to prevent these kind of merging issues in the future?

Reply:
Thanks. Merge conflicts are always a challenge. If it arose from a conflict flagged by git during a git merge – i.e., you did a 'git merge' and it left this file in conflict, and you then had to resolve it:

(1) For all but the simplest merge conflicts, I use a tool that helps you see the different versions. There are a number of such tools out there, but I personally like kdiff3. On a mac that uses the homebrew package manager, this can be installed with 'brew install kdiff3', then you can put the following in your ~/.gitconfig file:

```
[merge]
	tool = kdiff3
	conflictStyle = diff3

[mergetool "kdiff3"]
        path = /usr/local/bin/kdiff3
        trustExitCode = false
```

(there may be a bit more needed than that; it's been a while since I set it up). Then, when you have a conflict, you can run 'git mergetool path/to/file' and use that tool to help you resolve it. Of course, this involves learning another tool, and if you don't need to deal with merges too often, it may not be worth the effort. Whether or not you use kdiff3, you may want to add the 'conflictStyle' setting above: this makes it so that, when you're resolving a conflict, in addition to showing you the two versions in the file, it also shows you the common ancestor, which can make it easier to see what's going on (at least once you get used to looking at the conflict output). There's a bit more information about git conflicts here: https://git-scm.com/book/en/v2/Git-Branching-Basic-Branching-and-Merging

(2) Even with a tool like this, I find conflict resolution really error prone. So after I have resolved a conflict, I typically look carefully at the diffs of that file against either my branch, the master branch, or both to make sure it looks right.

Reply:
Thanks a lot for these suggestions, I will look into it!